templates/fish.txt

```diff
@@ -19,7 +19,11 @@ end
 # A copy of fish's internal cd function. This makes it possible to use
 # `alias cd=z` without causing an infinite loop.
 if ! builtin functions --query __zoxide_cd_internal
-    string replace --regex -- '^function cd\s' 'function __zoxide_cd_internal ' <$__fish_data_dir/functions/cd.fish | source
+    if status list-files functions/cd.fish >/dev/null
```

Comment:
this also needs `2>/dev/null` (could use the `&>/dev/null` shorthand),
else it will print an error on fish version 3